src/server/http/xsrf.js

```diff
@@ -4,18 +4,49 @@ export default function (kbnServer, server, config) {
   const disabled = config.get('server.xsrf.disableProtection');
   const versionHeader = 'kbn-version';
   const xsrfHeader = 'kbn-xsrf';
+  const contentTypeHeader = 'content-type';
+  const allowedRequestMediaTypes = ['application/json', 'application/x-ndjson'];
 
   server.ext('onPostAuth', function (req, reply) {
     if (disabled) {
       return reply.continue();
     }
 
     const isSafeMethod = req.method === 'get' || req.method === 'head';
-    const hasVersionHeader = versionHeader in req.headers;
-    const hasXsrfHeader = xsrfHeader in req.headers;
+    if (!isSafeMethod) {
```

Comment:
Rather than deeply nest the bulk of this function, how about inverting this conditional and returning early?

```js
if (isSafeMethod) {
  // no need to verify XSRF for GET or HEAD requests
  return;
}
```

It's not that big of a deal one way or another, but any time a conditional is more than a handful of lines, especially when it contains other conditionals, it stands out.